### ISSUE TITLE
Babel-eslint doesn't throw an error while parsing

### ISSUE THREAD

**Issue by dentuzhik** [babel/babel-eslint#476]
Guys from eslint instructed me to create a ticket here, the original report is in this issue: https://github.com/eslint/eslint/issues/8648

Version of `babel-eslint`: `7.2.3`

Please let me know, if I can provide any additional info.

**Comment by kentcdodds** [comment]
Based on the code sample you gave:

```javascript
var bar = function (foo) {
    const foo = 123;
    return foo;
};
```

That actually _is_ a syntax error. If I run this code in an environment that supports `const` (most environments these days) then I get:

<img width="473" alt="screen shot 2017-05-26 at 1 20 51 pm" src="https://cloud.githubusercontent.com/assets/1500684/26509491/3448fa10-4216-11e7-895f-366ba02156b4.png">

So the solution is to change your code so you're not redeclaring the parameter `foo`.

Unless I'm missing something. Please re-open this if there's more to it.

**Comment by dentuzhik** [comment]
Yes, I know that I should change my code, thank you.
But the question is why `babel-lint` doesn't throw an exception while parsing, while `espree` does?  

I would expect parser to fail when encountering error and be consistent with `espree`, since it makes sense that linting fails in this case (what currently simply doesn't happen).

**Comment by kentcdodds** [comment]
Ah, gotcha. My guess is this is actually a babylon parser issue. But that's outside of my level of experience. I'll re-open this and hopefully someone with more experience can chime in. Thanks for clarifying.

**Comment by soda0289** [comment]
There is an issue in babylon https://github.com/babel/babylon/issues/78

**Comment by hzoo** [comment]
Looks like this is just an issue in babel's parser ^ https://github.com/babel/babel/issues/6722
